plugins/copr.py

```diff
@@ -103,36 +104,88 @@ def set_argparser(parser):
-            if PY3:
-                self.copr_url = raw_config["copr-cli"].get("copr_url", None)
-            else:
-                self.copr_url = raw_config.get("copr-cli", "copr_url", None)
-            if self.copr_url != "https://copr.fedorainfracloud.org":
-                print(_("Warning: we are using non-standard Copr URL '{}'.").format(self.copr_url))
-
-
-        # Useful for forcing a distribution
         copr_plugin_config = ConfigParser()
-        config_file = None
+        config_files = []
         for path in self.base.conf.pluginconfpath:
-            test_config_file = '{}/{}.conf'.format(path, PLUGIN_CONF)
+            for filename in os.listdir('{0}/{1}.d'.format(path, PLUGIN_CONF)):
+                if filename.endswith('.conf'):
+                    config_file = os.path.join(path, PLUGIN_CONF + ".d", filename)
+                    config_files.append(config_file)
+
+            test_config_file = os.path.join(path, PLUGIN_CONF + ".conf")
             if os.path.isfile(test_config_file):
-                config_file = test_config_file
-
-        if config_file is not None:
-            copr_plugin_config.read(config_file)
-            if copr_plugin_config.has_option('main', 'distribution') and copr_plugin_config.has_option('main', 'releasever'):
-                distribution = copr_plugin_config.get('main', 'distribution')
-                releasever = copr_plugin_config.get('main', 'releasever')
-                self.chroot_config = [distribution, releasever]
-            else:
-                self.chroot_config = [False, False]
+                config_files.append(test_config_file)
+
+        project = []
+        try:
+            project = self.opts.arg[0]
+            project = project.split("/")
+        except (ValueError, IndexError):
+            pass
+
+        # Copr hub was not specified, using default hub `fedora`
+        if not self.opts.hub and len(project) != 3:
+            copr_hub = self.default_hub
+            self.copr_url = self.default_url
+
+        elif len(project) == 3 and self.opts.hub:
+            logger.critical(
+                _('Error: ') +
+                _('specify Copr hub either with `--hub` or using '
+                  '`copr_hub/copr_username/copr_projectname` format')
+            )
+            raise dnf.cli.CliError(_('multiple hubs specified'))
+
+        elif not config_files and self.opts.hub:
+            print(_("Warning: No configiguration file found, using the default instance of Copr."))
+            copr_hub = self.default_hub
+
+        # Copr hub URL should be specified in config file
+        elif self.opts.hub:
+            copr_hub = self.opts.hub
+
+        # Copr hub specified with hub/user/project format
+        elif len(project) == 3:
+            copr_hub = project[0]  # try to find hub in config files
+            self.copr_url = "https://" + project[0]  # otherwise use it directly as URL
+
+        if config_files:
+            hub_found = False
```

Comment:
The `config_files` is always array, so the `if config_files` is not needed;  you can iterate directly.

At the same time, I'd simply break the loop with as soon as possible (related config file found).